rpc/src/service_builder.rs

```diff
@@ -43,16 +52,17 @@ impl<'a> ServiceBuilder<'a> {
         min_fee_rate: FeeRate,
         reject_ill_transactions: bool,
     ) -> Self {
+        let rpc_method = PoolRpcImpl::new(
+            shared,
+            sync_shared_state,
+            min_fee_rate,
+            reject_ill_transactions,
+        )
+        .to_delegate();
         if self.config.pool_enable() {
-            self.io_handler.extend_with(
-                PoolRpcImpl::new(
-                    shared,
-                    sync_shared_state,
-                    min_fee_rate,
-                    reject_ill_transactions,
-                )
-                .to_delegate(),
-            );
+            self.io_handler.extend_with(rpc_method);
+        } else {
+            self.update_disabled_methods("Pool", rpc_method);
```

Comment:
Why not just fill the disabled methods with a handler which just returns the error message?

Reply:
I don't know what you mean

Reply:
```
for each method in rpc_method:
  hander.extend(method, |...| { return "Disabled" })
```

Reply:
This is also a solution, I can't decide which is better, but I chose middleware interception at the time

Reply:
I vote against middleware since it will have an overhead.

Reply:
ok, I'll change to add disable method